lib/psaux.js

```diff
@@ -1,6 +1,40 @@
 const execa = require('execa');
+const os = require('os');
 
-module.exports = options => {
+if (os.platform() != "win32") {
+  module.exports = getProcess
+} else {
+  module.exports = getWinProcess
+}
+
+function getWinProcess(options) {
+  return new Promise((resolve, reject) => {
+    execa('powershell', ['-noprofile',
+      '$perf = get-wmiobject -class Win32_PerfFormattedData_PerfProc_Process;$proc = get-wmiobject -class win32_process ;$TotalMem = get-wmiobject -class win32_computersystem | select -ExpandProperty TotalPhysicalMemory;$proc | foreach {$procID = $_;$procPerf = $perf | where {$_.creatingProcessID -eq $procID.ProcessID};$procID | select Name,@{n="User"; e= {$user = $_.getOwner(); if($user.user){"{0}\\{1}" -f $user.domain, $user.user}else {"System"}}},CommandLine,ProcessID,@{n="Memory (MB)";e={"{0:N3}" -f ($_.ws / 1mb)}},@{n="Memory (%)";e={"{0:N3}" -f ($_.ws / $TotalMem)}},@{n="StartTime";e={$_.converttoDateTime($_.CreationDate)}},@{n="CPU"; e={$item = $_;$COUNT = 0;if($procPerf.percentProcessorTime){$procPerf.PercentProcessorTime | foreach {$count += $_}; $percent = $count / $procPerf.PercentProcessorTime.length;"{0:N3}" -f [double]$percent} else {"{0:N3}" -f 0}}}} | ConvertTo-Json'
+    ]).then(result => {
+      var processes = JSON.parse(result.stdout);
+      processes = processes.reduce(parseWinProcesses, []);
+      processes.query = query;
+      resolve(processes);
+    });
+  });
+}
+
+function parseWinProcesses(list, ps) {
+  list.push({
+    user: ps["User"],
+    pid: ps["ProcessID"],
+    cpu: parseFloat(ps["CPU"]),
+    mem: parseFloat(ps["Memory (MB)"]),
+    started: ps["StartTime"],
+    command: ps["CommandLine"],
+    name: ps["Name"]
+  });
```

Comment:
We can get more, but i wasnt sure exactly what the other properties
translate to in windows terms.
On Feb 26, 2016 12:57 PM, "Hector Leon Zarco Garcia" <
notifications@github.com> wrote:

> In lib/psaux.js
> <https://github.com/zzarcon/psaux/pull/2#discussion_r54277481>:
>
> > +      processes.query = query;
> > +      resolve(processes);
> > +    });
> > +  });
> > +}
> > +
> > +function parseWinProcesses(list, ps) {
> > +  list.push({
> > +    user: ps["User"],
> > +    pid: ps["ProcessID"],
> > +    cpu: parseFloat(ps["CPU"]),
> > +    mem: parseFloat(ps["Memory (MB)"]),
> > +    started: ps["StartTime"],
> > +    command: ps["CommandLine"],
> > +    name: ps["Name"]
> > +  });
>
> This are all the properties we can get from windows? will we nice if we
> can also add any of those vsz, rss, tt, stat.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/zzarcon/psaux/pull/2/files#r54277481>.
>